### ISSUE TITLE
[BUG] no-custom-classname treats identifiers passed to clsx as class names

### ISSUE THREAD

**Issue by dimazuien** [francoismassart/eslint-plugin-tailwindcss#481]
**Describe the bug**

`tailwindcss/no-custom-classname` treats identifier arguments passed to `clsx` as literal class names.

For example, when `className` is passed to `clsx`, the rule reports `className` itself as an unknown Tailwind class, even though it is a JavaScript variable rather than a class name.

**To Reproduce**

1. Create a Tailwind CSS entry file:

```css
@import 'tailwindcss';
```

2. Configure ESLint:

```js
import tailwind from 'eslint-plugin-tailwindcss';

export default [
  {
    ...tailwind.configs.recommended,
    settings: {
      tailwindcss: {
        cssConfigPath: './src/globals.css',
      },
    },
  },
];
```

3. Lint the following file:

```js
import clsx from 'clsx';

const className = 'text-sm';

clsx('p-4', className);
```

4. Observe the warning:

```text
Classname 'className' is not a Tailwind CSS class
tailwindcss/no-custom-classname
```

**Expected behavior**

Literal strings and object keys passed to `clsx` should continue to be validated, but identifier arguments should not be treated as literal class names.

In the example above, the rule should not validate the text `className` as a Tailwind class.

**Screenshots**

Not applicable.

**Environment (please complete the following information):**

- OS: macOS 26.6.2
- Softwares + version used:
  - Node.js 24.18.0
  - pnpm 11.21.0
  - ESLint 9.39.4
  - eslint-plugin-tailwindcss 4.4.0
  - Tailwind CSS 4.3.1
  - clsx 2.1.1

**Additional context**

The behavior appears related to `parseKeyFunctions`. Its default configuration includes `clsx`, which is necessary for validating object keys:

```js
clsx({
  'text-sm': condition,
});
```

However, identifier arguments inside the same call are also interpreted as class-name candidates.

Removing `clsx` from `parseKeyFunctions` avoids the false positive but also disables useful validation of `clsx` object keys.

Individual identifier names can be added to the rule's `whitelist`, but that workaround is project-specific and requires anticipating every variable name that may be passed to `clsx`.

Related issue #267 concerns detecting object keys in functions such as `clsx`, but it describes a false negative and is already closed: https://github.com/francoismassart/eslint-plugin-tailwindcss/issues/267

**eslint config file or live demo**

The minimal configuration and reproduction are included above.
